gensim/models/phrases.py

```diff
@@ -157,19 +162,30 @@ def __str__(self):
             self.__class__.__name__, len(self.vocab), self.min_count,
             self.threshold, self.max_vocab_size)
 
-    @staticmethod
-    def learn_vocab(sentences, max_vocab_size, delimiter=b'_', progress_per=10000):
+    def learn_vocab(self, sentences):
         """Collect unigram/bigram counts from the `sentences` iterable."""
+        if not self.recode_to_utf8 and sentences is not None:
+            sentence = list(next(iter(sentences)))
```

Comment:
If I understand correctly, this will raise an exception if sentences is either an empty list or an empty generator.